sklearn/cluster/hierarchical.py

```diff
@@ -678,6 +678,12 @@ def __init__(self, n_clusters=2, affinity="euclidean",
         self.linkage = linkage
         self.affinity = affinity
         self.pooling_func = pooling_func
+        if self.pooling_func != np.mean:
```

Comment:
This should really be in fit, not in `__init__` according to our conventions. And ideally we'd change the default to something like 'deprecated' so that we can warn when the user explicitly passed `pooling_func=np.mean`

Reply:
Added the recommended changes, check once